File: Movies/C++/Theater.cpp
#include "Theater.hpp"
#include <string>
#include <vector>
#include <iostream>

Theater::Theater()
{}

Theater::~Theater()
{}

Theater::Theater(std::string name, std::string city, std::string state)
{
	this->name = name;
	this->city = city;
	this->state = state;
	this->sumFilm = 0;
	this->sumFood = 0;
}

std::string Theater::getName()
{
	return this->name;
}

std::string Theater::getCity()
{
	return this->city;
}

std::string Theater::getState()
{
	return this->state;
}

void Theater::addFilm(Movie* film)
{
	films.push_back(film);
}

void Theater::addMenu(Food* item)
{
	menu.push_back(item);
}

void Theater::addOrder(Food* order)
{
	orders.push_back(order);
}

void Theater::printFilms()
{
	std::cout << "***************************Movie Now Playing*************************\n";
	for(int i = 0; i < films.size(); i++)
	{
		std::cout << i+1 << ". " << films.at(i)->getTitle() << "  " << films.at(i)->getTime() << "  $" << films.at(i)->getPrice() << std::endl;
	}	
}

void Theater::printMenu()
{
	std::cout << "***************************Food Menu*************************\n";

	for(int i = 0; i < menu.size(); i++)
	{
		std::cout << i+1 << ". " << menu.at(i)->getName() << "  $" << menu.at(i)->getPrice() << std::endl;
	}
	
}


void Theater::printOrder()
{
	std::cout << "Your order consist of:\n";
	for(int i = 0; i < orders.size(); i++)
	{
		std::cout << i+1 << ". " << orders.at(i)->getName() << "  $" << orders.at(i)->getPrice() << "  Quant: " << orders.at(i)->getQuantity() << std::endl;
	}
}

double Theater::totalFilm()
{
	sumFilm = 0;
	for(int i = 0; i < films.size(); i++)
	{
		sumFilm += films.at(i)->getPrice();
	}
	return sumFilm;
	
}

double Theater::totalFood()
{
	sumFood = 0;
	for(int i = 0; i < orders.size(); i++)
	{
		sumFood = sumFood + (orders.at(i)->getPrice() * orders.at(i)->getQuantity());
	}
	return sumFood;
	
}

File: Movies/Python/Theater.py
from Movie import Movie 
from Black import Black
from Deadpool import Deadpool
from Marvel import Marvel
from Park import Park
from Mission import Mission
from Ocean import Ocean
from Sky import Sky
from Food import Food
from Soda import Soda
from Popcorn import Popcorn
from Snacks import Snacks
from Baskets import Baskets

class Theater:
	def __init__(self, name, city, state):
		self.name = name
		self.city = city
		self.state = state
		self.sumFood = 0
		self.films = []
		self.menu = []
		self.order = []
		
	def getName(self):
		return self.name
		
	def getCity(self):
		return self.city
		
	def getState(self):
		return self.state
		
	def addFilm(self, Movie):
		self.films.append(Movie)
	
	def addFood(self, Food):
		self.menu.append(Food)
		
	def addOrder(self, Food):
		self.order.append(Food)
	
	def printFilms(self):
		print "*****************Now Playing*****************"
		for x in range(len(self.films)):
			print "{}. {} {} ${} {}".format(x+1, self.films[x].getTitle(), self.films[x].getTime(), self.films[x].getPrice(), self.films[x].getType())

	def printMenu(self):
		print "*****************Food Menu*****************"
		for x in range(len(self.menu)):
			print "{}. {} ${}".format(x+1, self.menu[x].getName(), self.menu[x].getPrice())
			
	def printOrder(self):
		print "*****************Your Order*****************"
		for x in range(len(self.order)):
			print "{}. {} ${} quant: {}".format(x+1, self.order[x].getName(), self.order[x].getPrice(), self.order[x].getQuantity())

	def totalOrder(self):
		total = 0
		sumFood = 0
		for x in range(len(self.order)):
			total = self.order[x].getPrice() * self.order[x].getQuantity()
			sumFood += total
		return sumFood
		
	
